test/ha/autoscalerhpa_test.go

```diff
@@ -0,0 +1,85 @@
+// +build e2e
+
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ha
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/serving/pkg/apis/autoscaling"
+	rtesting "knative.dev/serving/pkg/testing/v1"
+	"knative.dev/serving/test"
+	"knative.dev/serving/test/e2e"
+	v1test "knative.dev/serving/test/v1"
+)
+
+const (
+	autoscalerHpaDeploymentName = "autoscaler-hpa"
+	autoscalerHpaLabel          = "app=autoscaler-hpa"
+)
+
+func TestAutoscalerHPAHANewRevision(t *testing.T) {
+	clients := e2e.Setup(t)
+
+	enableHA(t, clients, autoscalerHpaDeploymentName)
```

Comment:
Can we just patch the CM in the shell script that launches the test.
I think that's how we setup all the TLS tests, rather than patching inline
/cc @mattmoor

Reply:
yeah, I have been pushing back on folks changing the SUT from the test code.

Reply:
OK. I've moved changing the CM to the shell script. The Go test code is still scaling up and down the deployments. Stopping a Pod is essential to the test so scaling up and down the deployment is very close to that and I wouldn't pull it out of the Go code.